contracts/v0.8/utils/BigInts.sol

```diff
@@ -72,11 +73,15 @@ library BigInts {
     /// @return a int256 value and flog that indicates whether it was possible to convert or not (the value overflows int256 type)
     function toInt256(CommonTypes.BigInt memory value) internal view returns (int256, bool) {
         BigNumber memory max = BigNumbers.init(MAX_INT, false);
+        BigNumber memory min = BigNumbers.init(MIN_INT, true);
         BigNumber memory bigNumValue = BigNumbers.init(value.val, false);
-        if (BigNumbers.gt(bigNumValue, max)) {
+        if (BigNumbers.gt(bigNumValue, max) || BigNumbers.lt(bigNumValue, min)) {
             return (0, true);
         }
 
+        if (BigNumbers.eq(bigNumValue, min)) {
```

Comment:
Do we need this check?
Is it just for optimization?

Reply:
conversion uint256(bytes32(bigNumValue.val)) fails without this check